src/amqp.c

```diff
@@ -139,16 +144,18 @@ static bool on_receive(struct l_io *io, void *user_data)
 	routing_key = amqp_bytes_to_new_string(envelope.routing_key);
 	body = amqp_bytes_to_new_string(envelope.message.body);
 
-	err = amqp_ctx.read_cb(exchange, routing_key, body, user_data);
+	success = amqp_ctx.read_cb(exchange, routing_key, body, user_data);
+	if (!success)
+		/* FIXME: Put msg on the queue again */
```

Comment:
I think it is more like a TODO than a FIXME.
Also, I would make changes on the comment.

~Put~ Add **the** message...